File: day1/sum.py
total = 0
word_dict = {
    "one": "onee",
    "two": "twoo",
    "three": "three",
    "four": "fourr",
    "five": "fivee",
    "six": "sixx",
    "seven": "sevenn",
    "eight": "eightt",
    "nine": "ninee",
}

word_num_dict = {
    "one": "1",
    "two": "2",
    "thre": "3",
    "four": "4",
    "five": "5",
    "six": "6",
    "seven": "7",
    "eight": "8",
    "nine": "9",
}

with open("input.txt", "r") as f:
    for line in f:
        line = line.strip()
        for word, num in word_dict.items():
            line = line.replace(word, num)
        for word, num in word_num_dict.items():
            line = line.replace(word, num)
        if line:
            first_digit = next(char for char in line if char.isdigit())
            last_digit = next(char for char in reversed(line) if char.isdigit())
            total += int(first_digit + last_digit)

print(total)

File: day1/sum.vb
Imports System.IO
Imports System.Linq

Module Module1
    Sub Main()
        Dim total As Integer = 0
        Dim wordDict As New Dictionary(Of String, String) From {
            {"one", "onee"},
            {"two", "twoo"},
            {"three", "three"},
            {"four", "fourr"},
            {"five", "fivee"},
            {"six", "sixx"},
            {"seven", "sevenn"},
            {"eight", "eightt"},
            {"nine", "ninee"}
        }

        Dim wordNumDict As New Dictionary(Of String, String) From {
            {"one", "1"},
            {"two", "2"},
            {"thre", "3"},
            {"four", "4"},
            {"five", "5"},
            {"six", "6"},
            {"seven", "7"},
            {"eight", "8"},
            {"nine", "9"}
        }

        Dim lines As String() = File.ReadAllLines("input.txt")
        For Each line As String In lines
            Dim newLine As String = line
            For Each pair As KeyValuePair(Of String, String) In wordDict
                newLine = newLine.Replace(pair.Key, pair.Value)
            Next
            For Each pair As KeyValuePair(Of String, String) In wordNumDict
                newLine = newLine.Replace(pair.Key, pair.Value)
            Next
            If Not String.IsNullOrEmpty(newLine) Then
                Dim firstDigit As Char = newLine.FirstOrDefault(AddressOf Char.IsDigit)
                Dim lastDigit As Char = newLine.LastOrDefault(AddressOf Char.IsDigit)
                total += Integer.Parse(firstDigit.ToString()) + Integer.Parse(lastDigit.ToString())
            End If
        Next

        Console.WriteLine(total)
    End Sub
End Module